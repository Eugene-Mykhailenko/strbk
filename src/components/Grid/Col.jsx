import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Grid.scss';

const Col = (props) => {
  const { children, width, alignX, alignY, position } = props;

  const getClassnames = () =>
    classNames(
      styles.col,
      styles[`col-${width}`],
      styles[`justify-content-${alignX}`],
      styles[`align-items-${alignY}`],
      styles[`position-${position}`],
    );

  return <div className={getClassnames(styles)}>{children}</div>;
};

Col.propTypes = {
  /** Special property to pass children
   * elements directly into component output. */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  /** Column width. */
  width: PropTypes.string,
  /** This property specifies the default alignment
   * on X-axis for items inside the column. */
  alignX: PropTypes.string,
  /** This property specifies the default alignment
   * on Y-axis for items inside the column. */
  alignY: PropTypes.string,
  /** Property specifies the type of positioning column. */
  position: PropTypes.string,
};

Col.defaultProps = {
  children: undefined,
  width: undefined,
  alignX: 'start',
  alignY: 'center',
  position: undefined,
};

export default Col;
