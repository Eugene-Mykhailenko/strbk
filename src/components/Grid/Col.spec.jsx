import React from 'react';
import Col from './Col';

const div = <div className="find-me">Lorem Ipsum is simply dummy text</div>;

describe('Col', () => {
  it('should render with no children by default', () => {
    const col = shallow(<Col />);

    expect(col.find('.col')).toHaveLength(1);
    expect(col).toMatchSnapshot();
  });

  it('should render provided children', () => {
    const col = shallow(<Col>{div}</Col>);

    expect(col.find('.find-me')).toHaveLength(1);
    expect(col).toMatchSnapshot();
  });

  it('should be able to justify content on X-axis', () => {
    const col = shallow(<Col>{div}</Col>);

    col.setProps({ alignX: 'center' });

    expect(col.prop('className')).toMatch(/justify-content-center/);
    expect(col).toMatchSnapshot();

    col.setProps({ alignX: 'end' });

    expect(col.prop('className')).toMatch(/justify-content-end/);
    expect(col).toMatchSnapshot();
  });

  it('should be able to justify content on Y-axis', () => {
    const col = shallow(<Col>{div}</Col>);

    col.setProps({ alignY: 'start' });

    expect(col.prop('className')).toMatch(/align-items-start/);
    expect(col).toMatchSnapshot();

    col.setProps({ alignY: 'end' });

    expect(col.prop('className')).toMatch(/align-items-end/);
    expect(col).toMatchSnapshot();
  });

  it('should be able to align itself on X-axis', () => {
    const col = shallow(<Col>{div}</Col>);

    col.setProps({ position: 'left' });

    expect(col.prop('className')).toMatch(/position-left/);
    expect(col).toMatchSnapshot();

    col.setProps({ position: 'right' });

    expect(col.prop('className')).toMatch(/position-right/);
    expect(col).toMatchSnapshot();

    col.setProps({ position: 'center' });

    expect(col.prop('className')).toMatch(/position-center/);
    expect(col).toMatchSnapshot();
  });
});
