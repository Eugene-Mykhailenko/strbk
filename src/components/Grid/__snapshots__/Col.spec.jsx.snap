// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should be able to align itself on X-axis 1`] = `
<div
  className="col col-undefined justify-content-start align-items-center position-left"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should be able to align itself on X-axis 2`] = `
<div
  className="col col-undefined justify-content-start align-items-center position-right"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should be able to align itself on X-axis 3`] = `
<div
  className="col col-undefined justify-content-start align-items-center position-center"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should be able to justify content on X-axis 1`] = `
<div
  className="col col-undefined justify-content-center align-items-center position-undefined"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should be able to justify content on X-axis 2`] = `
<div
  className="col col-undefined justify-content-end align-items-center position-undefined"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should be able to justify content on Y-axis 1`] = `
<div
  className="col col-undefined justify-content-start align-items-start position-undefined"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should be able to justify content on Y-axis 2`] = `
<div
  className="col col-undefined justify-content-start align-items-end position-undefined"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should render provided children 1`] = `
<div
  className="col col-undefined justify-content-start align-items-center position-undefined"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Col should render with no children by default 1`] = `
<div
  className="col col-undefined justify-content-start align-items-center position-undefined"
/>
`;
